### PR TITLE
Add IsSqlFragment implicit for better error messages

### PR DESCRIPTION
**Pull request by a-morales**
@jdegoes This implements #26

### CODE REVIEW

**Review by jdegoes — APPROVED** [review]

**Comment by jdegoes** [comment]
@a-morales I think we need to make the type parameter for `IsSqlFragment` contravariant in order to make this compile.

**Comment by a-morales** [comment]
@jdegoes I've added the contra variant to the `IsSqlFragment` trait.

**Comment by jdegoes** [comment]
@a-morales Thank you! Please just fix the merge commits, and if it passes the CI build, I will be happy to merge. Thanks for your work on this!

File: core/src/test/scala/zio/jdbc/IsSqlFragmentSpec.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package zio.jdbc
+
+import zio.test._
+import zio.test.Assertion._
+
+object IsSqlFragmentSpec extends ZIOSpecDefault {
+  def spec: ZSpec[Environment with TestEnvironment, Any] =
+    suite("IsSqlFragment") {
+      test("Mapped fragments that are 'complete' compile") {
+        val result = typeCheck {
```
Comment (jdegoes) [inline core/src/test/scala/zio/jdbc/IsSqlFragmentSpec.scala:10]: Love the tests of compile-ness here!

**Comment by jdegoes** [comment]
@a-morales Sorry for the trouble, but there are a few more conflicts. Can you resolve? Looks good to merge to me!

**Comment by a-morales** [comment]
@jdegoes I've merged in the latest main branch, fixed the conflicts and merged the commits into a single one. This is ready to merge once tests pass.

**Pull request merged**